### PR TITLE
introduce anyhow, move away from toposort

### PR DESCRIPTION
**Pull request by drahnr**
A coalescation of changes which is yet to be split into three individual ones:

* [x] `fn toposort` detects cycles in graphs when there are none, a diamond is not a cycle. Using `fn kosaraju_scc` instead, which provides the same as well as the cycles which helps debugging
* [x] move to `anyhow` <del>(or `eyre`, TBD)</del> - the error handling before was not really intuitive
* [x] write a <del>`graphml` (or </del>`graphviz`<del>, TBD)</del> to file<del>, this has to be refactored into a subcommand<del> as a flag to various sub-commands, where it made sense

The motivation behind this was a major PITA releasing `juice` on the weekend which has such diamond shaped dependency graph, but cargo-unleash did not provide any info on where the supposedly existent cycles are in the dependency tree.

### CODE REVIEW

**Comment by drahnr** [comment]
Defering the impl of the `graphviz` generation due to the plethora of libs that do partial impls, `petgraph-graphml` is far from perfect without digging deep into spec, the commits should be atomic and reviewable.

**Comment by drahnr** [comment]
Verifying the cycles, it shows that the detected ones are bogus. This leads me a bit in the direction that the graph is actually constructed not quite right.

**Comment by drahnr** [comment]
> Verifying the cycles, it shows that the detected ones are bogus. This leads me a bit in the direction that the graph is actually constructed not quite right.

This was an issue that got introduced in the first commit, extracting the `Workspace`, but the follow up closure `maybe_patch` which removes the `DevDependencies` section, was hence not read. The last commit resolves this and no more cycles are detected.

**Review by gnunicorn — CHANGES_REQUESTED** [review]
Very nice, thanks! The dot-visualisations are a nice new features!

Just two request:
1. do you mind adding a test case showing the problem and that it is resolved with this PR? I honestly am not the best crack re graphiz and want to make sure that we have a test case ensuring we don't introduce regressions upon this later!
2. please add this and the new dot-visualisation-feature to the Changelog! That's soo cool!

**Comment by drahnr** [comment]
Tests are bit messy, but there doesn't seem to be any released cargo test helper create that is released. Changelog was updated :)

**Comment by drahnr** [comment]
And rebased.

File: src/commands/to_release.rs [hunk 1]
```diff
@@ -103,17 +126,293 @@ where
...
+                pkg.version().to_string().as_str()
+            )
+        },
+    );
+    dest.write_all(format!("{:?}", &dot).as_bytes())?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (gnunicorn) [inline src/commands/to_release.rs:196]: we usually use actual fixtures and cli-e2e tests for cases this complicated, but this is fine, too...!

**Review by gnunicorn — APPROVED** [review]
Nice! thanks!

**Pull request merged**